packages/blockchain-extension/extension/commands/deployCommand.ts

```diff
@@ -0,0 +1,61 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+'use strict';
+import * as vscode from 'vscode';
+import {PackageRegistryEntry} from '../registries/PackageRegistryEntry';
+import {FabricEnvironmentRegistryEntry, IFabricEnvironmentConnection, LogType} from 'ibm-blockchain-platform-common';
+import {ExtensionCommands} from '../../ExtensionCommands';
+import {FabricEnvironmentManager} from '../fabric/environments/FabricEnvironmentManager';
+import {VSCodeBlockchainOutputAdapter} from '../logging/VSCodeBlockchainOutputAdapter';
+
+export async function deploySmartContract(requireCommit: boolean, fabricEnvironmentRegistryEntry: FabricEnvironmentRegistryEntry, ordererName: string, channelName: string, peerNames: Array<string>, smartContractName: string, smartContractVersion: string, sequenceNumber: number, chosenPackage: PackageRegistryEntry): Promise<void> {
+    const outputAdapter: VSCodeBlockchainOutputAdapter = VSCodeBlockchainOutputAdapter.instance();
+    outputAdapter.log(LogType.INFO, 'Deploy Smart Contract');
+    try {
+        await vscode.window.withProgress({
+            location: vscode.ProgressLocation.Notification,
+            title: 'IBM Blockchain Platform Extension',
+            cancellable: false
+        }, async (progress: vscode.Progress<{ message: string }>) => {
+
+            let packageId: string;
+
+            await vscode.commands.executeCommand(ExtensionCommands.CONNECT_TO_ENVIRONMENT, fabricEnvironmentRegistryEntry);
+            const connection: IFabricEnvironmentConnection = FabricEnvironmentManager.instance().getConnection();
+            if (!connection) {
+                // something went wrong with connecting so return
+                return;
+            }
+
+            progress.report({message: `Installing Smart Contract on peer`});
+            packageId = await vscode.commands.executeCommand(ExtensionCommands.INSTALL_SMART_CONTRACT, peerNames, chosenPackage);
+            if (!packageId) {
+                throw new Error('Package was not installed. No packageId was returned');
+            }
+
+            progress.report({message: `Approving Smart Contract`});
+            await vscode.commands.executeCommand(ExtensionCommands.APPROVE_SMART_CONTRACT, ordererName, channelName, peerNames, smartContractName, smartContractVersion, packageId, sequenceNumber);
+
+            if (requireCommit) {
+                progress.report({message: `Committing Smart Contract`});
+                await vscode.commands.executeCommand(ExtensionCommands.COMMIT_SMART_CONTRACT, ordererName, channelName, peerNames, smartContractName, smartContractVersion, sequenceNumber);
+            }
+
+            outputAdapter.log(LogType.SUCCESS, 'Successfully deployed smart contract');
+        });
+
+    } catch (error) {
```

Comment:
Is it worth us being more specific about where it fails? Just thinking from a debugging perspective, it might be nice if it was to say `"Failed deploying smart contract at 'approving' stage: ${error.messsage}"` as an example.

Reply:
Although I guess the command which throws the error has a message like this already - nevermind!

Reply:
you will get other pop ups from command it failed on